### PR TITLE
Migrate Sequence (CTCGreedyDecoder-1, CTCGreedyDecoderSeqLen-6, GRUCe…

### PR DESCRIPTION
**Pull request by steve-y**
Migrate Sequence (CTCGreedyDecoder-1, CTCGreedyDecoderSeqLen-6, GRUCell-3, LSTMCell-4, OneHot-1, RNNCell-1) and Sort (ExperimentalDetectronTopKROIs-6)

### Details:
 - *Create template plugin reference tests for target operation: CTCGreedyDecoder-1, CTCGreedyDecoderSeqLen-6, GRUCell-3, LSTMCell-4, OneHot-1, RNNCell-1, ExperimentalDetectronTopKROIs-6*
 - *Remove ngraph backend files*
 - *Add visitor/op for CTCGreedyDecoder-1, CTCGreedyDecoderSeqLen-6, ExperimentalDetectronTopKROIs-6*
 - *Update CMakeLists.txt*
 - *Add R"(.\*ReferenceGRUCellTestHardsigmoidActivationFunction.\*gru_cell_hardsigmoid_activation_function)" to skip_tests_config.cpp*

### Tickets:
 - *63944*
 - *64099*
 - *64066*
 - *64079*
 - *64036*
 - *63978*
 - *64104*

### CODE REVIEW

File: docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            false,
+            Tensor(IN_ET, {2, 2, 1, 1}, std::vector<T>{1.0f, 1.0f, 0.0f, -1.0f}),
+            "ctc_greedy_decoder_multiple_sequences"),
+    };
+    return params;
+}
+
+std::vector<CTCGreedyDecoderParams> generateCombinedParams() {
+    const std::vector<std::vector<CTCGreedyDecoderParams>> generatedParams {
+        generateParams<element::Type_t::bf16>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder.cpp:117]: In the operation semantic I don't see any limitations for integer or other types. Could we extend the set of data types?

File: docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder_seq_len.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            Tensor(element::i32, {3}, std::vector<int32_t>{2, 3, 1}),
+            "evaluate_ctc_greedy_decoder_seq_len_multiple_batches2"),
+    };
+    return params;
+}
+
+std::vector<CTCGreedyDecoderSeqLenParams> generateCombinedParams() {
+    const std::vector<std::vector<CTCGreedyDecoderSeqLenParams>> generatedParams {
+        generateParams<element::Type_t::bf16>(),
+        generateParams<element::Type_t::f16>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder_seq_len.cpp:181]: the same

File: docs/template_plugin/tests/functional/op_reference/experimental_detectron_topkrois.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            2,
+            Tensor(ET, {2, 4}, std::vector<T>{13.0f, 10.0f, 18.0f, 15.0f, 3.0f, 2.0f, 7.0f, 9.0f}),
+            "experimental_detectron_topk_rois_eval"),
+    };
+    return params;
+}
+
+std::vector<ExperimentalDetectronTopKROIsParams> generateCombinedParams() {
+    const std::vector<std::vector<ExperimentalDetectronTopKROIsParams>> generatedParams {
+        generateParams<element::Type_t::bf16>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/experimental_detectron_topkrois.cpp:93]: the same

File: docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder.cpp (hunk 1 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder.cpp:117]: In https://docs.openvino.ai/latest/openvino_docs_ops_sequence_CTCGreedyDecoder_1.html, input and output types are float point.

File: docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder_seq_len.cpp (hunk 2 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/ctc_greedy_decoder_seq_len.cpp:181]: In https://docs.openvino.ai/latest/openvino_docs_ops_sequence_CTCGreedyDecoderSeqLen_6.html, input type is floating point.

File: docs/template_plugin/tests/functional/op_reference/experimental_detectron_topkrois.cpp (hunk 3 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/experimental_detectron_topkrois.cpp:93]: In https://docs.openvino.ai/latest/openvino_docs_ops_sort_ExperimentalDetectronTopKROIs_6.html, input and output types are floating point.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**